pkg/watch/health.go

```diff
@@ -85,19 +93,25 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 // service it is monitoring. Every HEALTHCHECK_INTERVAL it
 // performs a health check and writes that information to
 // consul
-func (p *PodWatch) MonitorHealth(node string, store kp.Store, shutdownCh chan bool) {
+func (p *PodWatch) MonitorHealth(store kp.Store, node string, shutdownCh chan bool) {
 	for {
 		select {
 		case <-time.After(HEALTHCHECK_INTERVAL):
-			p.checkHealth(node, p.manifest.StatusPort, store)
+			p.checkHealth(store, node, p.manifest.StatusPort)
 		case <-shutdownCh:
 			return
 		}
 	}
 }
 
-func (p *PodWatch) checkHealth(node string, port int, store kp.Store) {
-	resp, err := kp.HttpStatusCheck(node, port)
+func (p *PodWatch) checkHealth(store kp.Store, node string, port int) {
+	var resp *http.Response
+	var err error
+	if p.manifest.StatusHTTP == true {
+		resp, err = kp.HttpsStatusCheck(p.client, node, port)
```

Comment:
This has gotten too unwieldy. The code is passing around a node name, a port, an HTTP client, and a manifest just so it can have the data available right here to perform the status check it needs.  This state comes from a mix of fields in a struct definition and parameters threaded through half a dozen functions.

I want you consider an alternate way of structuring this check: define a new `StatusCheck` type that encapsulates all the state it needs to perform the check and pass around an instance of that.  The following might work

    type StatusCheck struct {
        ID string
        Node string // maybe?
        URI string
        Client *http.Client
    }

    func (sc StatusCheck) Check() health.Result { ... }

You wouldn't need quite so many parameters in all your functions, and it decouples the _gathering_ of a check from the _handling_ of one.